src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -351,8 +356,37 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
-
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            bool hasFormatCulture = false;
+            string enUsCultureName = "en-US";
+            CultureInfo formatCulture = culture;
+            string revFormatStr = formatString;
+
+            if (!string.IsNullOrEmpty(formatString) && formatString.StartsWith("[$-", StringComparison.OrdinalIgnoreCase))
+            {
+                int startIdx = formatString.IndexOf("[$-", StringComparison.Ordinal) + 3;

```

Comment:
>formatString.IndexOf("[$-" [](http://example.com/codeflow?start=31&length=26)

We do  this check twice. (StartsWith does it) 

Can we just do it once? 
Just call IndexOf, and then check StartIndex==0.

Reply:
updated here